### ISSUE TITLE
break out tap into a standalone lit package

### ISSUE THREAD

**Issue by nihildeb** [luvit/luvit#728]
I really like using tap, and I've used it quite a bit in my sqlite lit package. I tore a couple files out of luvit, and still there are deps on 'luvit/utils' for colorize, and whatever the exports object is. I'm still trying to figure out how exports is initialized so that I can use it in a lit module just to get tap and utils all wired up properly. 

Just thought, it would be great if I could just do something like
```
luvi . -m /tests/run.lua
``` 
in my **lit package** and have it just work, without adding dependencies to the project, just that it's all in a <project_dir>/tests/ dir

**Comment by creationix** [comment]
What do you mean "without adding dependencies"?  I agree having the tap package in lit is a great idea.  There is no reason it needs luvit's utils, it can work just fine using pretty-print directly which most luvi apps have already.

**Comment by creationix** [comment]
As far as the exports, add `luvit/require` as a dependency to your luvi app and it will have luvit style require semantics.  This is highly recommended since most packages on lit assume these semantics.

If you want to fake it on your own, look at how exports is implemented in `luvit/require`: https://github.com/luvit/luvit/blob/master/deps/require.lua#L243-L263

**Comment by nihildeb** [comment]
"without adding dependencies": well it would be nice and easy to have it as a lit package, and just add the dep and then lit install and yay tests, but then I can also imagine that a package like luvit has 30 dependencies, and if they all had tests, then we'd have 30x luvit/tap in the binary. Now that I think of it though, lit doesn't work like that like npm does with really deep dependancy trees does it...

In any case, tap is great, thanks for that. I will remove 'utils' and add require and pretty_print and see how it goes.

**Comment by nihildeb** [comment]
I could probably make a package out of tap and manage it, but it's your code coming straight out of luvit, so I wouldn't do that unless you wanted me to...

**Comment by creationix** [comment]
I'll publish it for you, I'm in a meeting at the moment, so after that.

**Comment by nihildeb** [comment]
thanks. btw, it's tied up with utils and exports because I pulled it out of luvit I think, and there is a util deep_equal which is exported. it would be great to keep this extensibility in there to write customer asserts and such. I added my own assert eq method to the exports in there, but it wasn't available in my tests. Didn't take the time to figure out how it should work, but it would be nice if it had :)
